cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaMetadataStateManagerMockTest.java

```diff
@@ -0,0 +1,609 @@
+                .withReplicas(3)
+                .withNewJbodStorage()
+                    .withVolumes(new PersistentClaimStorageBuilder().withId(0).withSize("100Gi").build())
+                .endJbodStorage()
+                .withRoles(ProcessRoles.CONTROLLER)
+                .withResources(new ResourceRequirementsBuilder().withRequests(Map.of("cpu", new Quantity("4"))).build())
+            .endSpec()
+            .build();
+    private final static KafkaNodePool BROKERS = new KafkaNodePoolBuilder()
+            .withNewMetadata()
+                .withName("brokers")
+                .withNamespace(NAMESPACE)
+                .withLabels(Map.of(Labels.STRIMZI_CLUSTER_LABEL, CLUSTER_NAME))
+            .endMetadata()
+            .withNewSpec()
+                .withReplicas(3)
+                .withNewJbodStorage()
+                    .withVolumes(new PersistentClaimStorageBuilder().withId(0).withSize("200Gi").build())
+                .endJbodStorage()
+                .withRoles(ProcessRoles.BROKER)
+                .withResources(new ResourceRequirementsBuilder().withRequests(Map.of("cpu", new Quantity("6"))).build())
+            .endSpec()
+            .build();
+    private static final List<KafkaPool> POOLS = NodePoolUtils.createKafkaPools(Reconciliation.DUMMY_RECONCILIATION, KAFKA, List.of(CONTROLLERS, BROKERS), Map.of(), Map.of(), true, SHARED_ENV_PROVIDER);
+
+    /**
+     * Tests migration from Zookeeper State to KRaftPostMigration State.
+     * The strimzi.io/kraft annotation is set to `migration`
+     *
+     * @param context Test context
+     */
+    @Test
+    public void testMigrationFromZooKeeperToKRaftPostMigrationState(VertxTestContext context) {
+
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_KRAFT, "migration")
+                .endMetadata()
+                .withNewStatus()
+                    .withKafkaMetadataState(ZooKeeper.name())
+                .endStatus()
+                .build();
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        SecretOperator mockSecretOps = supplier.secretOperations;
+        Secret secret = new Secret();
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.secretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+
+        MockKafkaReconciler kr = new MockKafkaReconciler(
+                new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+                vertx,
+                CONFIG,
+                supplier,
+                new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+                kafka,
+                List.of(BROKERS, CONTROLLERS),
+                createKafkaCluster(KAFKA),
```

Comment:
it's interesting that this doesn't have any impact on the tests but, you are passing `KAFKA` to the `createKafkaCluster` instead of the local variable `kafka`. And that's the same across all the tests. You should fix it.

Reply:
It was because even though I was passing this parameter here, in the constructor It was not being used since I had seperate `createKafkaCluster(kafka)` there which pointed to correct kafka resource.  I have fixed that now